widget_driver/doc/drivers_without_generation.md

```diff
@@ -0,0 +1,60 @@
+# Drivers Without Generation
+
+You do not need to use code generation to use `WidgetDrivers`. You can just as well create all the needed code manually. It is just a bit more manual work for you 😅
+
+It is actually not so much code you have to create for each `driver`. But as you project grows it still adds up.
+
+*So the recommendation would still be to use code generation.*
+
+## What do I need to do
+
+You just need two extra classes!  
+Except for the actual `WidgetDriver` you need to create a `TestDriver` version of your `Driver`. And you need to create a `WidgetDriverProvider` which can create an instance for your `driver` and `testDriver`.
+
+### Creating the TestDriver
+
+The `TestDriver` is used when your widget is created during widgetTests. It cannot have any dependencies to any instance of something. It should just contain hard coded values. *(It can contain static values from some other type.)*
+
+Your `TestDriver` needs to extend the `TestDriver` type and implement your `Driver` type. Here is an example of how to define it:
+
+```dart
+// Imagine you have a driver called `MyCoolWidgetDriver`
+
+class MyCoolWidgetTestDriver extends TestDriver
+    implements MyCoolWidgetDriver {
+  @override
+  String get coolName => 'A cool name';
+
+  @override
+  String get coolImageUrl => CoolImageService.testCoolImageUrl;
```

Comment:
```suggestion
  String get coolImageUrl => "https://picsum.photos/300/300";
```
I would avoid using another type in this example, just to emphasis that your test driver should *not* depend on other stuff and for most cases the best would be to use hardcoded values. You could also go with any other static URL, like https://github.githubassets.com/images/modules/logos_page/GitHub-Mark.png

Reply:
I just wanted to highlight here that you can use static variables also.. And that you don't have to have the actual value hardcoded in-place.

For example, say that you have a `driver` which has a property that returns a list of some objects. Something like this:

```dart
@DriverProperty([MyCoolObject(value1: "123", value2: "456"), MyCoolObject(value1: "abc", value2: "def")])
List<MyCoolObject> get theCoolObjects  => ...
```

Then maybe it would be nicer to define that like this:

```dart
@DriverProperty([MyCoolObject.testInstance1, MyCoolObject.testInstance2])
List<MyCoolObject> get theCoolObjects  => ...

// or 

@DriverProperty(MyCoolObject.listOfTestInstances)
List<MyCoolObject> get theCoolObjects  => ...
```

And have some helper/extension on `MyCoolObject` which defines these static test instances..